File: leet-code/771. Jewels and Stones/JewelsAndStones771_4.java
package com.example;

import java.util.Map;
import java.util.HashMap;

public class JewelsAndStones771_4 {
    public int numJewelsInStones(String jewels, String stones) {
        Map<Character, Integer> stoneMap = new HashMap<>();
        for (char ch : stones.toCharArray()) {
            stoneMap.put(ch, stoneMap.getOrDefault(ch, 0) + 1);
        }
        int count = 0;
        for (char ch : jewels.toCharArray()) {
            if (stoneMap.containsKey(ch)) {
                count += stoneMap.get(ch);
            }
        }
        return count;
    }
}

File: leet-code/771. Jewels and Stones/JewelsAndStones771_4.kt
class JewelsAndStones771_4 {
    fun numJewelsInStones(jewels: String, stones: String): Int {
        val set: MutableSet<Char> = HashSet()
        for (j in jewels) {
            set.add(j)
        }
        var cnt: Int = 0;
        for (s in stones) {
            if (set.contains(s)) {
                cnt += 1
            }
        }
        return cnt
    }
}